share/smack/top.py

```diff
@@ -304,6 +304,13 @@ def arguments():
                 (note that memory-safety is the union of valid-deref,
                 valid-free, memleak)''')
 
+    translate_group.add_argument(
```

Comment:
I am wondering if we should merge various encoding options we currently have, into one option that contains several choices. We can all the option `--encoding-of-ints` or something like that, and the options would be `integer, bit-precise, bit-precise-pointers, and wrapped-integer`. The default would be integer. That would make it cleaner since currently you can select multiple of these options which does not make sense.

Reply:
Good idea. Let me implement it.

Reply:
Good suggestion! But can all of these be mix and matched!?
We should try this (and maybe have a regression that tries all possible combinations).

Reply:
It seems the current implementation forces `--bit-precise-pointers` to imply `--bit-precise`. We should lift the restriction.

Reply:
Please see #600